src/flattenizer.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Flattens an object
+ *
+ * @param {Object} obj - the object to unflatten
+ * @returns {Object} - the flattened object, empty if provided object is undefined
+ */
+export const flatten = (obj) => {
+   let flattened = {};
+
+    for (let key in obj) {
+        if (!obj.hasOwnProperty(key))
+            continue;
+
+        if (typeof obj[key] === 'object') {
+            let flatObject = flatten(obj[key]);
+            for (let subKey in flatObject) {
+                // append to create new key value and assign it's value
+                flattened[key + '.' + subKey] = flatObject[subKey];
```

Comment:
That was my reaction as well, but I thought the "dot" was awfully hard to see. If string interpolation is used, I would document the key format in a comment directly above